common/src/main/java/com/wynntils/screens/WynntilsQuestBookScreen.java

```diff
@@ -0,0 +1,513 @@
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.screens.widgets.BackButton;
+import com.wynntils.screens.widgets.PageSelectorButton;
+import com.wynntils.screens.widgets.QuestBookSearchWidget;
+import com.wynntils.screens.widgets.QuestButton;
+import com.wynntils.screens.widgets.ReloadQuestsButton;
+import com.wynntils.screens.widgets.TextInputBoxWidget;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.utils.Pair;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.model.questbook.QuestBookManager;
+import com.wynntils.wynn.model.questbook.QuestInfo;
+import com.wynntils.wynn.model.questbook.QuestStatus;
+import com.wynntils.wynn.objects.ProfessionInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.components.AbstractButton;
+import net.minecraft.client.gui.components.Widget;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+public class WynntilsQuestBookScreen extends Screen implements SearchableScreen {
+    private static final int QUESTS_PER_PAGE = 13;
+    private static final List<Component> RELOAD_TOOLTIP = List.of(
+            new TranslatableComponent("screens.wynntils.wynntilsQuestBook.reload.name").withStyle(ChatFormatting.WHITE),
+            new TranslatableComponent("screens.wynntils.wynntilsQuestBook.reload.description")
+                    .withStyle(ChatFormatting.GRAY));
+
+    private Widget hovered = null;
+    private final QuestBookSearchWidget searchWidget;
+    private int currentPage = 0;
+    private int maxPage = 0;
+    private List<QuestInfo> quests;
+    private List<QuestButton> questButtons = new ArrayList<>();
+    private QuestInfo tracked = null;
+
+    public WynntilsQuestBookScreen() {
+        super(new TranslatableComponent("screens.wynntils.wynntilsQuestBook.name"));
+
+        this.searchWidget = new QuestBookSearchWidget(
+                (int) (Texture.QUEST_BOOK_BACKGROUND.width() / 2f + 15),
+                0,
+                Texture.QUEST_BOOK_SEARCH.width(),
+                Texture.QUEST_BOOK_SEARCH.height(),
+                this::updateQuestsFilter,
+                this);
+
+        WynntilsMod.getEventBus().register(this);
```

Comment:
I looks a bit "unbalanced" to register the event bus in the constructor, but unregister in `onClose`. Maybe register in `init` instead?

Reply:
In fact, why is `searchWidget` created in the constructor instead of in `init`?

Reply:
Aha.

Reply:
Maybe add what you just said as a comment! :)